src/service.rs

```diff
@@ -55,7 +55,7 @@ impl Display for ExitedError {
 #[derive(Debug)]
 pub struct LspService<S> {
     inner: Router<S, ExitedError>,
-    state: Arc<ServerState>,
+    pub(crate) state: Arc<ServerState>,
```

Comment:
@mrnossiom is this the thing you're referring to [when you said](https://github.com/tower-lsp/tower-lsp/issues/12) this PR "changes public API"?

I'm not a Rust expert, but my understanding is that `pub(crate)` doesn't affect anything outside of this crate, so this doesn't feel like a public API change to me.

Reply:
this does not change public API indeed

the API change is `serve` which is not generic over `tower::Service`s anymore